File: ruby_related/naive_queue.rb
### Write the class
class Queue

  def initialize
    # begin with a new array
    @queue = []
  end

  def push(val)
    # push val to back
    #   of list
   @queue.push(val)
    # return self
    self
  end

  def pop
    # return a val at the
    #   front of the list
    @queue.shift
  end

end

File: javascripts/naive_queue.js
// Write the constructor
function Queue() {
  // a list attribute
  var queue = [];
} 

Queue.prototype.push = function(val){
  // push a val into the back of the list
  this.queue.push(val);
  // return this
  return this;
};

Queue.prototype.pop = function(){
  // remove a val from the front of the list
  return this.queue.shift();
};

